masonry_core/src/core/contexts.rs

```diff
@@ -1178,6 +1197,80 @@ impl_context_method!(
             self.global_state.mutate_callbacks.push(callback);
         }
 
+        /// Get direct reference to the stored child, and a context handle for that child.
+        ///
+        /// This context lets you set pass flags for the child widget, which may be tricky.
+        /// In general, you should avoid setting the flags of a pass that runs before the
+        /// pass you're currently in.
+        ///
+        /// See [pass documentation](crate::doc::doc_05_pass_system) for the pass order.
+        pub fn get_raw<Child: Widget + FromDynWidget + ?Sized>(
+            &mut self,
+            child: &mut WidgetPod<Child>,
+        ) -> (&Child, RawCtx<'_>) {
+            let child_state_mut = self
+                .widget_state_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_mut = self
+                .widget_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_properties = self
+                .properties_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_ctx = RawCtx {
+                global_state: self.global_state,
+                parent_widget_state: self.widget_state,
+                widget_state: child_state_mut.item,
+                widget_state_children: child_state_mut.children,
+                widget_children: child_mut.children,
+                properties_children: child_properties.children,
+            };
+
+            let widget = Child::from_dyn(&**child_mut.item).unwrap();
+
+            (widget, child_ctx)
+        }
+
+        /// Get direct mutable reference to the stored child, and a context handle for that child.
+        ///
+        /// This context lets you set pass flags for the child widget, which may be tricky.
+        /// In general, you should avoid setting the flags of a pass that runs before the
+        /// pass you're currently in.
+        ///
+        /// See [pass documentation](crate::doc::doc_05_pass_system) for the pass order.
+        pub fn get_raw_mut<Child: Widget + FromDynWidget + AllowRawMut + ?Sized>(
+            &mut self,
+            child: &mut WidgetPod<Child>,
+        ) -> (&mut Child, RawCtx<'_>) {
```

Comment:
Can this return a `WidgetMut` to the child instead? That way, `RawCtx` could become read only, preserving encapsulation.

Reply:
> I could maybe go the other way around and make get_raw() return a QueryCtx, though.

Done.

Returning a QueryCtx means a bunch of context types now need to carry a `&DefaultProperties` around, which isn't a huge cost.